eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/Recovery.scala

```diff
@@ -152,30 +199,26 @@ private class Acceptor(endpoint: ReplicationEndpoint) extends Actor {
 
   private val recovery = new Recovery(endpoint)
 
-  def initializing: Receive = {
+  def initializing: Receive = recovering orElse {
     case Process =>
       context.become(processing)
-    case Recover(links, promise) =>
-      println(s"[recovery of ${endpoint.id}] Checking replication progress with remote endpoints ...")
-      context.become(recoveringMetadata(context.actorOf(Props(new RecoveryManager(endpoint.id, links))), promise))
   }
 
-  def recoveringMetadata(recoveryManager: ActorRef, promise: Promise[Unit]): Receive = {
-    case re: ReplicationReadEnvelope if re.incompatibleWith(endpoint.applicationName, endpoint.applicationVersion) =>
-      sender ! ReplicationReadFailure(IncompatibleApplicationVersionException(endpoint.id, endpoint.applicationVersion, re.targetApplicationVersion), re.payload.targetLogId)
-    case re: ReplicationReadEnvelope =>
-      recoveryManager forward re
-    case MetadataRecoveryCompleted =>
-      endpoint.connectors.foreach(_.activate())
+  def recovering: Receive = {
+    case Recover(links, promise) =>
+      endpoint.connectors.foreach(_.activate(Some(links.map(_.remoteLogId))))
+      val recoveryManager = context.actorOf(Props(new RecoveryManager(endpoint.id, links)))
       context.become(recoveringEvents(recoveryManager, promise) orElse processing)
+    case RecoveryFinished =>
+      context.become(processing)
   }
 
   def recoveringEvents(recoveryManager: ActorRef, promise: Promise[Unit]): Receive = {
     case writeSuccess: ReplicationWriteSuccess =>
       recoveryManager forward writeSuccess
     case EventRecoveryCompleted =>
       promise.success(())
-      context.become(processing)
+      context.become(recovering orElse processing)
```

Comment:
Just for the `links` provided in the `Recover` message, but there might be several `Recover` _invocations_ (one for filtered, one for unfiltered connections). Recovery is complete when `RecoveryFinished` is received.